File: tools/_strip_bom.js
const fs = require('fs');
const path = require('path');

const gui = 'c:\\Users\\nemex\\OneDrive\\Documenti\\GitHub\\ocr-support-patch\\ocr_support_compatibility_pach\\gui';
const files = ['window_culture.gui', 'window_military.gui', 'window_faith.gui', 'window_inventory.gui'];
const BOM = Buffer.from([0xEF, 0xBB, 0xBF]);
const results = [];

for (const name of files) {
    const filePath = path.join(gui, name);
    const data = fs.readFileSync(filePath);
    const hasBOM = data.length >= 3 && data[0] === 0xEF && data[1] === 0xBB && data[2] === 0xBF;

    if (hasBOM) {
        const noBom = data.slice(3);
        fs.writeFileSync(filePath, noBom);
        results.push(`STRIPPED: ${name} (${data.length} -> ${noBom.length})`);
    } else {
        results.push(`NO-BOM: ${name} (${data.length})`);
    }
}

// Verify
results.push('\n--- VERIFICA ---');
for (const name of files) {
    const filePath = path.join(gui, name);
    const data = fs.readFileSync(filePath);
    const hasBOM = data.length >= 3 && data[0] === 0xEF && data[1] === 0xBB && data[2] === 0xBF;
    results.push(`${name}: ${hasBOM ? 'ERRORE BOM!' : 'OK senza BOM'} (${data.length} bytes)`);
}

const outPath = 'c:\\Users\\nemex\\OneDrive\\Documenti\\GitHub\\ocr-support-patch\\tools\\_bom_results.txt';
fs.writeFileSync(outPath, results.join('\n') + '\n', 'utf8');

File: tools/_strip_bom.py
"""Strip UTF-8 BOM from specified GUI files."""
import os, sys

GUI_DIR = r'c:\Users\nemex\OneDrive\Documenti\GitHub\ocr-support-patch\ocr_support_compatibility_pach\gui'

FILES = [
    'window_culture.gui',
    'window_military.gui',
    'window_faith.gui',
    'window_inventory.gui',
]

BOM = b'\xef\xbb\xbf'

for name in FILES:
    path = os.path.join(GUI_DIR, name)
    with open(path, 'rb') as f:
        data = f.read()
    if data[:3] == BOM:
        with open(path, 'wb') as f:
            f.write(data[3:])
        print(f'STRIPPED: {name} ({len(data)} -> {len(data)-3} bytes)')
    else:
        print(f'NO-BOM:  {name} ({len(data)} bytes)')

# Verify
print('\n--- VERIFICA ---')
for name in FILES:
    path = os.path.join(GUI_DIR, name)
    with open(path, 'rb') as f:
        head = f.read(3)
    has_bom = head == BOM
    size = os.path.getsize(path)
    status = 'ERRORE: BOM ancora presente!' if has_bom else 'OK: senza BOM'
    print(f'{name}: {status} ({size} bytes)')
